maint.tf

```
@@ -0,0 +1,77 @@
+module "state_bucket" {
+  source        = "git::https://github.com/plus3it/terraform-aws-tardigrade-s3-bucket.git?ref=5.0.0"
+  bucket        = var.backend_config.bucket
+  force_destroy = var.backend_config.force_destroy
+  versioning    = var.backend_config.versioning
+
+  public_access_block = {
+    block_public_acls       = var.backend_config.public_access_block.block_public_acls
+    block_public_policy     = var.backend_config.public_access_block.block_public_policy
+    ignore_public_acls      = var.backend_config.public_access_block.ignore_public_acls
+    restrict_public_buckets = var.backend_config.public_access_block.restrict_public_buckets
+  }
+
+  policy = {
+    json = jsonencode({
+      "Version" : "2012-10-17",
+      "Statement" : [
+        {
+          "Sid" : "EnforcedTLS",
+          "Effect" : "Deny",
+          "Principal" : "*",
+          "Action" : "s3:*",
+          "Resource" : [
+            "arn:${data.aws_partition.current.partition}:s3:::${var.backend_config.bucket}",
+            "arn:${data.aws_partition.current.partition}:s3:::${var.backend_config.bucket}/*"
+          ],
+          "Condition" : {
+            "Bool" : {
+              "aws:SecureTransport" : "false"
+            }
+          }
+        },
+        {
+          "Sid" : "RootAccess",
+          "Effect" : "Allow",
+          "Principal" : {
+            "AWS" : "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:root"
+          },
+          "Action" : "s3:*",
+          "Resource" : [
+            "arn:${data.aws_partition.current.partition}:s3:::${var.backend_config.bucket}",
+            "arn:${data.aws_partition.current.partition}:s3:::${var.backend_config.bucket}/*"
+          ]
+        },
+        {
+          "Sid" : "DenyS3DeleteObject",
+          "Action" : [
+            "s3:DeleteObject"
+          ],
+          "Effect" : "Deny",
+          "Resource" : [
+            "arn:${data.aws_partition.current.partition}:s3:::${var.backend_config.bucket}/*"
+          ],
+          "Principal" : "*"
+        }
+      ]
+    })
+  }
+
+
+  server_side_encryption_configuration = {
+    bucket_key_enabled = var.backend_config.server_side_encryption_configuration.bucket_key_enabled
+    sse_algorithm      = var.backend_config.server_side_encryption_configuration.sse_algorithm
+    kms_master_key_id  = var.backend_config.server_side_encryption_configuration.kms_master_key_id
+  }
```

Similar simplification here:

```suggestion
  server_side_encryption_configuration = var.backend_config.server_side_encryption_configuration
```